packages/http/README.md

```diff
@@ -400,6 +400,12 @@ You would get this in response
 ]
 ```
 
+## Prism Decision Flow Diagram
+
+The below diagram represents all logical decision we make to figure out the best HTTP response to server. Note that this diagram is valid only when mocking is turned on.
+
+![Decision Flow Diagram](docs/images/pdfd.png)
```

Comment:
Not sure why github is not showing that image in the preview... seems that the file isn't uploaded to the right location. I expect that to be resolved once merged.

Reply:
What does `pdfd` stand for?

Reply:
prism decision flow diagram

Reply:
haha could we maybe call it mock-server-dfd?

Reply:
maybe it needs a ./ at the front?

Reply:
> oooo wait I found one incorrect thing. Only one though!
> 
> After negotiator block it says "Success?" then the No goes up to "takes lowest 2xx" but 2xx is for success. What are we trying to say here?

I meant: if the negotiator block resulted in "success" message then X otherwise "go up and... take lowest 2xx". 

And this is actually a correct behaviour. We first try to match to a given code and if we fail we try to match to lowest 2xx. This part of the diagram represents a recurrence in the code.

Reply:
@philsturgeon ^